h2o-algos/src/main/java/hex/tree/TreeHandler.java

```diff
@@ -0,0 +1,172 @@
+package hex.tree;
+
+import hex.genmodel.algos.tree.SharedTreeNode;
+import hex.genmodel.algos.tree.SharedTreeSubgraph;
+import hex.schemas.TreeV3;
+import water.MemoryManager;
+import water.api.Handler;
+import water.util.ArrayUtils;
+import water.util.CollectionUtils;
+
+import java.util.*;
+
+/**
+ * Handling requests for various model trees
+ */
+public class TreeHandler extends Handler {
+    private static final int NO_CHILD = -1;
+
+    public TreeV3 getTree(final int version, final TreeV3 args) {
+        validateArgs(args);
+        final SharedTreeModel model = (SharedTreeModel) args.model.key().get();
+        if (model == null) throw new IllegalArgumentException("Unknown tree key: " + args.model.toString());
+
+        final SharedTreeModel.SharedTreeOutput sharedTreeOutput = (SharedTreeModel.SharedTreeOutput) model._output;
+        final CompressedTree auxCompressedTree = sharedTreeOutput._treeKeysAux[args.tree_number][args.tree_class].get();
+        final SharedTreeSubgraph sharedTreeSubgraph = sharedTreeOutput._treeKeys[args.tree_number][args.tree_class].get()
+                .toSharedTreeSubgraph(auxCompressedTree, sharedTreeOutput._names, sharedTreeOutput._domains);
+
+        final CompressedTreeFormat compressedTreeFormat = convertSharedTreeSubgraph(sharedTreeSubgraph);
```

Comment:
I think to name this "CompressedTreeFormat" clashes with "CompressedTree", confusing in my opinion.

Reply:
Renamed.